course04/homework00/ciaranfrain/sorted_array_tests.cpp

```diff
@@ -0,0 +1,132 @@
+#include "sorted_array_tests.h"
+
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
+const int SortedArrayTests::mDefaultMaxSize = 10;
+
+SortedArrayTests::SortedArrayTests(): mSortedArray(mDefaultMaxSize)
+{}
+
+void SortedArrayTests::RunTests()
+{
+	std::cout << "Running tests..." << std::endl;
+
+	RunTest("ShouldInitialiseEmpty",             std::bind(&SortedArrayTests::ShouldInitialiseEmpty, this));
+	RunTest("ShouldInsertOneElement",            std::bind(&SortedArrayTests::ShouldInsertOneElement, this));
+	RunTest("ShouldRetrieveCorrectElementValue", std::bind(&SortedArrayTests::ShouldRetrieveCorrectElementValue, this));
+	RunTest("ShouldThrowOnExceedingSize",        std::bind(&SortedArrayTests::ShouldThrowOnExceedingSize, this));
+	RunTest("ShouldInsertLesserValueAtFront",    std::bind(&SortedArrayTests::ShouldInsertLesserValueAtFront, this));
+	RunTest("ShouldInsertGreaterValueAtBack",    std::bind(&SortedArrayTests::ShouldInsertGreaterValueAtBack, this));
+	RunTest("ShouldSortElements",                std::bind(&SortedArrayTests::ShouldSortElements, this));
```

Comment:
Nice and neat, well done

Reply:
In general we use lambdas instead of std::bind, but thats minor. std::bind is pre c++11